### PR TITLE
Add basic Spark SQL tests and more support for SQLContext methods.

### PR DESCRIPTION
**Pull request by chetmancini**
This gets basic SQL usage down.

Currently thinking through what a great `DataFrame` API might feel like in Clojure. Lots of room for improvement there. Of course, a lot of the `flambo.api` RDD functions work for `DataFrame` out of the box (`first`, `count`...)

### CODE REVIEW

File: src/clojure/flambo/sql.clj [hunk 1]
```diff
@@ -4,28 +4,66 @@
...
+
+(defn ^DataFrame table
+  "Return a table as a DataFrame"
+  [sql-context table-name]
+  (.table sql-context table-name))
+
+(defn table-names
+  "Return a seq of strings of table names, optionally within a specific database"
+  ([sql-context]
+   (seq (.tableNames sql-context)))
```
Comment (jeffreyksmithjr) [inline src/clojure/flambo/sql.clj]: Is there a reason to prefer a seq over the array returned by `.tableNames`?

**Comment by senor-hadoop** [comment]
Looks like there is some duplication with [pull request 60](https://github.com/yieldbot/flambo/pull/60).

**Comment by chetmancini** [comment]
@senor-hadoop ah, I see. Great stuff. I'd be happy to rebase/clean up once yours gets in.

File: src/clojure/flambo/sql.clj [hunk 2]
```diff
@@ -4,28 +4,66 @@
...
+  "Load a JSON file (one object per line) "
+  [sql-context path]
   (.jsonFile sql-context path))
 
 (defn register-data-frame-as-table [sql-context df table-name]
   (.registerDataFrameAsTable sql-context df table-name))
 
-(defn cache-table [sql-context table-name]
+(defn json-rdd
+  "Load an RDD of JSON strings (one object per line), inferring the schame, and returning a DataFrame"
```
Comment (senor-hadoop) [inline src/clojure/flambo/sql.clj]: Small typo: schame -> schema

**Comment by sorenmacbeth** [comment]
@chetmancini I've merged #60, so if you want to rebase and clean up this one, I can merge it after that.

**Comment by chetmancini** [comment]
@sorenmacbeth Thanks. Rebased and cleaned up.

**Comment by chetmancini** [comment]
Anything more I should change to get this merged in?

**Comment by sorenmacbeth** [comment]
LGTM

**Pull request merged**